### DISCUSSION TITLE
No .env file found.

### DISCUSSION THREAD

**Discussion by dm9tr0** [dani-garcia/vaultwarden#2192] [Q&A]
Clean installation, according to instruction:
```
docker pull vaultwarden/server:latest
docker run -d --name vaultwarden -v /vw-data/:/data/ -p 80:80 vaultwarden/server:latest

```
What should be inside of this .env file? Why it did not create automatically?

**Comment by Teitei011** [comment]
Have you find the answer? I'm having the same problem

**Comment by th12289** [comment]
I do not have a solution for this, but with docker-compose i had only 1 issue and got it running.
https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose

I had to change the following line

    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile:ro

to

    volumes:
      - Caddyfile:/etc/caddy/ #Caddyfile:ro

If the container is running, put in the mentionend Caddyfile and restart.

--------------------------

Dont forget to set the admin token, to login and create users in the docker-compose.yaml.

    environment:
      - ADMIN_TOKEN=YOURFULLSECRETADMINTOKEN

**Reply by swotai** [reply]
How come you comment out the caddyfile:ro part?
`  - ./Caddyfile:/etc/caddy/Caddyfile:ro` This maps the Caddyfile under same path as the docker-compose.yaml into a read only file in /etc/caddy
`  - Caddyfile:/etc/caddy/ #Caddyfile:ro` which is essentially `  - Caddyfile:/etc/caddy/` is wonky, you're pointing a file as a directory in the specification.

**Comment by RealOrangeOne** [comment]
The env file should be something matching https://github.com/dani-garcia/vaultwarden/blob/main/.env.template.

The reason it's not created automatically is because it's optional. The idea is that vaultwarden needs some environment variables setting for it to run. How you set those is up to you (and may vary depending on how the application is run). For convenience, you can specify the environment variables in an "env file". The template I linked above shows which variables you may want to set, and what they do.

The lack of env file is not an error, so long as you set the variables it needs _somehow_. If it says there's no env file, that's just a warning.

**Comment by BlackDex** [comment]
It even is an info message.
And indeed no issue not being there.

**Comment by remram44** [comment]
I ran into this.

Maybe the installation instructions should be update if env values are required? Currently the README says:

```
docker pull vaultwarden/server:latest
docker run -d --name vaultwarden -v /vw-data/:/data/ -p 80:80 vaultwarden/server:latest
```

I see that issues are being thrown over the discussions so I'm not going to bother trying to contribute.

**Reply by BlackDex** [reply]
Vaultwarden is able to, and not mandatory to, read an `.env` file from the same path where you start the application. The `.env` is mostly used for non docker environments. If you use docker, the you probably use the `-e` option. That's a whole different item.

Also, it's an INFO message, not a warning, error or panic.

And, issues which are not application issues don't belong on the GitHub issues section but here in discussions, so yea those items will be transferred. Also, discussion items are more visible then closed issues, that is also a reason i transfer some items.

If you want to update the wiki and improve it, go ahead everyone can update it, no commits, or PR's needed.
